### DISCUSSION TITLE
Tutorial: Creating a mega menu with the block system

### DISCUSSION THREAD

**Discussion by justintadlock** [WordPress/developer-blog-content#182] [Topic Ideas]
First, I want to say that this is a _REALLY_ rough idea. One of the downfalls of the block system (really, even classic theming) is that there are no easy ways to create mega menus. And if I had a nickel for every time a developer asked me about this, I could at least buy myself a dinner at a fancy restaurant.

I'm just throwing this topic out here to talk ideas/solutions at the moment. It's not formed well enough to approve it as a topic on the blog. Think of this as more of an open invitation to answer:

- Are mega menus possible with the block system?
- If so, can we build something that's user friendly with the tools at hand?
- If so, can we explain this through a tutorial?

I've heard some ideas tossed around using patterns or template parts. I'm going to try to explore some options through code in January, but I also want to make sure this is wide open for feedback/discussion.

**Comment by bph** [comment]
Please continue discussion on the issue. 
- #197
